solutions/CF Packmen.mdx

```diff
@@ -0,0 +1,295 @@
+---
+id: cf-packmen
+source: CF
+title: Packmen
+author: Kevin Sheng
+---
+
+(No official editorial)
+
+Binary search on the minimum time it will take for all the Packmen to eat all the pellets.
+For each Packman, we can greedily calculate the maximum amount of pellets that it can eat in the given amount of time and then remove those pellets.
+However, each Packman must eat all of the uneaten food to its left, or else all the food can't be eaten in the given time. Assigning other Packmen to the uneaten pellets would be pointless, because they would be farther away and thus have no way of reaching the uneatable pellet.
+
+Note that there are two main cases: One where the Packman first goes to the right then to the left, and another where the Packman first goes to the *left* then the right. We calculate the food eaten in both of these cases and take the maximum. Note that there is also a special case where a Packman doesn't have any pellets to the left, so it can go to the right, eating as many pellets as time allows.
+
+**Time Complexity:** $\mathcal{O}(N\log N)$
```

Comment:
Move this to the top of the editorial

Reply:
Just the "two main cases" part of the time complexity?

Reply:
Just the time complexity line